### PR TITLE
check for http method strings in endpoint instruction factory

### LINKED ISSUE #78: Add checks for request methods
**Is your feature request related to a problem? Please describe.**
While HTTP request methods are a finite set, the `methods` attribute in the `structure.endpoint` function is annotated as `list[str]`. 

Humans can make mistakes, and with blurred eyes, it might be challenging to quickly identify these mistakes, even in a well-formatted log message like this:

```python
cadwyn.exceptions.RouterGenerationError: Endpoint "['PATH'] /v1xxx/{some_id}/fewfew" you tried to restore in "EnablingObsoleteRoutes" wasn't among the deleted routes
```

**Describe the solution you'd like**
I propose annotating the `methods` attribute of the `structure.endpoint` function with a Literal.Additionally, incorporating checks with raising expressive error messages for this case would enhance the user experience.

### PR DESCRIPTION
**Pull request by m0n0x41d**

### CODE REVIEW

**Review by zmievsa — CHANGES_REQUESTED** [review]
Thanks for your PR!
Please, update the changelog and fix the comments.

**Comment by zmievsa** [comment]
Please, use proper names for both the branch and the pull request. There is no need to add issue name to your pull request. Instead, you should attach PR to issue in the development section by hand.
image

File: tests/test_structure.py [hunk 1]
```diff
@@ -409,3 +410,14 @@ class B(A):
...
+
+def test__endpoint_instruction_factory_interface__with_wrong_http_methods__should_raise_error():
+    with pytest.raises(
+        LintingError,
+        match=re.escape(
+            "The following HTTP methods are not valid: DEATH, STRAND."
+            "Please use valid HTTP methods such as GET, POST, PUT, PATCH, DELETE, OPTIONS, HEAD.",
+        ),
+    ):
+        endpoint("/test", ["DEATH", "STRAND"], func_name="Nofunc")
```
[the remark this answers is not in the document]
Comment (m0n0x41d) [reply tests/test_structure.py]: done here!

**Review by zmievsa — DISMISSED** [review]
Great job. Thanks!

**Review by zmievsa — APPROVED** [review]

**Pull request merged**